### PR TITLE
feat(waypoints): start avoiding edge overlap on shape

### PR DESCRIPTION
**Pull request by tbouffard**
Add several high level tests to demonstrate for edge overlap on shape.
The "vacation request from bonita" also shows the behavior.

### Screenshots

**Info**: first screenshot is the rendering prior this PR

#### svg

vacation_request_v01_svg_01_before_change
vacation_request_v01_svg_02_without_overlapping

#### bpmn

vacation_request_v01_bpmn_01_before_change

vacation_request_v01_bpmn_02_without_overlapping

### Notes

Some overlap still remain. This PR only avoid overlap when the original edge should have been horizontal.

For instance, when involving Bottom Left to Top Right (waypoints-avoid-edge-overlap-03-elements_in_front.bpmn.xml test diagram)

image

Cycle Top Right to Bottom Left neither (waypoints-positions-cycle_02_gateways_in_cycle.bpmn.xml)

image

### Refactoring needed

This PR requires refactoring as the `WaypointsComputer` manages too much things (both positioning and actual points computation) and contains duplications.
I suggest to do the refactoring in a dedicated PR to avoid introducing too much changes here.

### CODE REVIEW

File: java/src/main/java/io/process/analytics/tools/bpmn/generator/converter/waypoint/GridSearcher.java [hunk 1]
```diff
@@ -69,4 +69,30 @@ public boolean isShapeExistAbove(final Position positionFrom) {
                 .anyMatch(p -> p.getY() == positionFrom.getY() - 1);
     }
 
+
+    public BendConfiguration searchEmptyRow(Position positionFrom, Position positionTo, BendDirection searchDirection) {
```
Comment (tbouffard) [inline java/src/main/java/io/process/analytics/tools/bpmn/generator/converter/waypoint/GridSearcher.java:73]: the name doesn't describe what the method does. I will refactor this in a new PR

File: java/src/test/java/io/process/analytics/tools/bpmn/generator/converter/waypoint/WayPointsComputerTest.java [hunk 2]
```diff
@@ -12,33 +12,187 @@
...
+        //  |         elt        |
+        //  |         elt        |
+        //  | from    elt     to |
+        //  |         elt        |
+        //  +--------------------+
+        EdgeDirection edgeDirection = computeEdgeDirection(position(1,3), position(4,3),
+                Grid.of(position(3, 1), position(3, 2), position(3, 3), position(3, 4)));
+        assertThat(edgeDirection.direction).isEqualTo(LeftToRight);
+        assertThat(edgeDirection.orientation).isEqualTo(VerticalHorizontalVertical);
+        assertBendConfigurationOnBottom(edgeDirection.bendConfiguration, 2);
```
Comment (baptistemesta) [inline java/src/test/java/io/process/analytics/tools/bpmn/generator/converter/waypoint/WayPointsComputerTest.java:76]: Nice tests :)
